contrib/managed-azure/setup_dev_environment.md

```diff
@@ -0,0 +1,279 @@
+KV_NAME=<KV_NAME>
+AZURE_DISK_SP_NAME="azure-disk-<PERSONAL_SP_NAME>"
+AZURE_FILE_SP_NAME="azure-file-<PERSONAL_SP_NAME>"
+NODEPOOL_MGMT="nodepool-mgmt-<PERSONAL_SP_NAME>"
+CLOUD_PROVIDER_SP_NAME="cloud-provider-<PERSONAL_SP_NAME>"
+CNCC_NAME="cncc-<PERSONAL_SP_NAME>"
+CONTROL_PLANE_SP_NAME="cpo-<PERSONAL_SP_NAME>"
+IMAGE_REGISTRY_SP_NAME="ciro-<PERSONAL_SP_NAME>"
+INGRESS_SP_NAME="ingress-<PERSONAL_SP_NAME>"
+USER_ACCOUNT_ID=<ID_OF_USER_ACCOUNT_OR_SP>
+SP_OUTPUT_FILE=<OUTPUT_FILE_FOR_CREATED_SPs>
+AKS_CLUSTER_RG_NAME=<AKS_CLUSTER_RG_NAME>
+DNS_RECORD_NAME=<DNS_RECORD_NAME>
+EXTERNAL_DNS_SP_NAME=<EXTERNAL_DNS_SP_NAME>
+EXTERNAL_DNS_CREDS=<PATH_TO_FILE_WITH_DNS_CREDS>
+DNS_ZONE_NAME="<DNS_RECORD_NAME>.hypershift.azure.devcluster.openshift.com"
+PARENT_DNS_ZONE="hypershift.azure.devcluster.openshift.com"
+PARENT_DNS_RG="os4-common"
+HC_NAME=<HC_NAME>
+```
+
+## Steps
+Note: Steps 1-7 set up the environment so that if created in a persistent group they can be 
+reused for creation of clusters in the future.
+
+### 1. Retrieve User Account Details
+
+```sh
+ACCOUNT_DETAILS=$(az account show --query '{subscriptionId: id, tenantId: tenantId}' -o json)
+SUBSCRIPTION_ID=$(echo "$ACCOUNT_DETAILS" | jq -r '.subscriptionId')
+TENANT_ID=$(echo "$ACCOUNT_DETAILS" | jq -r '.tenantId')
+```
+
+### 2. Create Service Principal for Authentication
+
+```sh
+SP_DETAILS=$(az ad sp create-for-rbac --name "$PERSONAL_SP_NAME" --role Contributor --scopes "/subscriptions/$SUBSCRIPTION_ID" -o json)
+CLIENT_ID=$(echo "$SP_DETAILS" | jq -r '.appId')
+CLIENT_SECRET=$(echo "$SP_DETAILS" | jq -r '.password')
+
+cat <<EOF > $SP_AKS_CREDS
+{
+  "subscriptionId": "$SUBSCRIPTION_ID",
+  "tenantId": "$TENANT_ID",
+  "clientId": "$CLIENT_ID",
+  "clientSecret": "$CLIENT_SECRET"
+}
+EOF
+```
+
+### 3. Create Managed Identities for AKS Cluster Creation
+
+```sh
+az identity create --name $AKS_CP_MI_NAME --resource-group $PERSISTENT_RG_NAME
+az identity create --name $AKS_KUBELET_MI_NAME --resource-group $PERSISTENT_RG_NAME
+```
+
+```shell
+export AKS_CP_MI_ID=$(az identity show --name $AKS_CP_MI_NAME --resource-group $PERSISTENT_RG_NAME --query id -o tsv)
+export AKS_KUBELET_MI_ID=$(az identity show --name $AKS_KUBELET_MI_NAME --resource-group $PERSISTENT_RG_NAME --query id -o tsv)
+```
+
+### 4. Create Key Vault for Certificate Storage
+
+```sh
+az keyvault create --name $KV_NAME --resource-group $PERSISTENT_RG_NAME --location $LOCATION --enable-rbac-authorization
+az role assignment create --assignee ${USER_ACCOUNT_ID} --scope /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${PERSISTENT_RG_NAME}/providers/Microsoft.KeyVault/vaults/${KV_NAME} --role "Key Vault Administrator"
+```
+
+### 5. Create Service Principals
+
+```sh
+cloudProvider=$(az ad sp create-for-rbac --name "${CLOUD_PROVIDER_SP_NAME}" --create-cert --cert "${CLOUD_PROVIDER_SP_NAME}" --keyvault "${KV_NAME}" --query "{clientID: appId, certificateName: '${CLOUD_PROVIDER_SP_NAME}'}" -o json)
```

Comment:
Just double checking, this automatically gets filled out correctly?

Reply:
yep